integr8ly/integreatly-operator
# refactor of creation of missing metrics alerts

# Issue link
https://issues.redhat.com/browse/MGDAPI-4964

#Changes

- openshift-monitoring - rhoam missing metrics alert with metrics used in critical alerts in that stack for rhoam with appropriate time 
- rhoam absent metrics monitoring - rhoam missing metrics alerts with metrics used in crtical alerts in that stack for rhoam with appropriate time 
- threescale absent metrics monitoring - threescale missing metrics created at the same time as the metrics critical to those alerts.
- isinreconciling error should no longer use absent.

- moving custom domain alerts to threescale reconciler but should be present even when custom domain isn't configure

# Verification steps

Run in a ROSA STS enabled cluster.
Verify that there is new alert rhoam missing metrics in openshift-monitoring.
Verify that on installation the alert doesn't fire. reach out to Tomas 

Trigger the metrics missing alert for both openshift-monitoring and observability promethues by scaling down RHOAM.

Non custom domain
Ensure that the observability rhoam missing metric alert does not include rhoam_custom_domain metric in the query

Custom domain

Configure Custom domain - probably will need an uninstall and install again 
Ensure that the observabiity rhoam missing metric alert does contain the rhoam_custom_domain metric in the query.
